### PR TITLE
Added WitnessScriptPubKeyV1 for sending to Taproot addresses

### PR DESCRIPTION
**Pull request by nkohen**
Not usable for spending from Taproot addresses, just sending.

### CODE REVIEW

**Review by benthecarman — APPROVED** [review]
LGTM maybe worth adding a test

File: core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala [hunk 1]
```diff
@@ -1337,6 +1340,27 @@ object P2WSHWitnessSPKV0 extends ScriptFactory[P2WSHWitnessSPKV0] {
...
+
+object WitnessScriptPubKeyV1 extends ScriptFactory[WitnessScriptPubKeyV1] {
+
+  override def fromAsm(asm: Seq[ScriptToken]): WitnessScriptPubKeyV1 = {
+    buildScript(asm.toVector,
+                WitnessScriptPubKeyV1.apply,
+                s"Given asm was not a P2WSHWitnessSPKV0, got $asm")
+  }
+
+  def isValidAsm(asm: Seq[ScriptToken]): Boolean = {
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/protocol/script/ScriptPubKey.scala:1369]: nit: `override`

File: core/src/main/scala/org/bitcoins/core/wallet/utxo/InputInfo.scala [hunk 2]
```diff
@@ -319,11 +322,12 @@ object InputInfo {
                                 witness,
                                 conditionalPath,
                                 hashPreImages)
-      case wspk: UnassignedWitnessScriptPubKey =>
+      case wspk @ (_: UnassignedWitnessScriptPubKey |
+          _: WitnessScriptPubKeyV1) =>
```
Comment (Christewart) [inline core/src/main/scala/org/bitcoins/core/wallet/utxo/InputInfo.scala:326]: any reason to not just do a separate case here and throw the `UnsupportedException`?

**Review by Christewart — COMMENTED** [review]
LGTM, but like ben said a test case would be nice. I guess this would mean generating a taproot address on bitcoind and then seeing if we can construct transaction that spends to that taproot address? 

I guess this would mean manually constructing the tx? I'm not sure what the best test case could be. There was some chatter on the bitcoin dev mailing list about how to best test taproot wallets, and it appears this isn't straight forward with bitcoind is my understanding: 
https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-October/019533.html

@benthecarman what are your thoughts?

**Comment by benthecarman** [comment]
> LGTM, but like ben said a test case would be nice. I guess this would mean generating a taproot address on bitcoind and then seeing if we can construct transaction that spends to that taproot address?
> 
> I guess this would mean manually constructing the tx? I'm not sure what the best test case could be. There was some chatter on the bitcoin dev mailing list about how to best test taproot wallets, and it appears this isn't straight forward with bitcoind is my understanding: [lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-October/019533.html](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2021-October/019533.html)
> 
> @benthecarman what are your thoughts?

There are taproot addresses in the bech32m tests, could just try to send to one of those (just convert it to regtest)

**Pull request merged**